docs/consumer/Converter.md

```diff
@@ -0,0 +1,114 @@
+# Stake Converter
+
+The Stake Converter is on the consumer side and is connected to an External Staker on the Provider side.
+This handles the normalization of the external tokens and _converts_ them into "Virtual Stake".
+There is a 1:1 connection between a Converter and a [Virtual Staking Contract](./VirtualStaking.md)
+which handles the actual issuance.
+The converter is connected to the Provider chain via IBC and handles the various packets coming from it.
+
+**TODO** This was called the "Consumer contract" in previous versions and probably needs a better name
+
+## Setup
+
+When we [deploy the contracts](../ibc/Overview.md#deployment), we connect the Stake Converter on the consumer chain
+with an [External Staking](../provider/ExternalStaking.md) contract on the Provider. Once this connection is established,
+Consumer governance can authorize this Stake Converter with some ability to mint on the ["Virtual Staking" contract](./VirtualStaking.md).
+
+Note that when we deploy the Stake Converter, we configure the address of the "virtual staking" contract. We must also
+define a price oracle contract on setup. (see ["Price Normalization"](#price-normalization) below)
+
+## Staking Flow
+
+Once the connection is established, the provider can send various "virtual stake" messages to the converter, which is responsible
+for processing them and normalizing for the local "virtual staking" module.
+
+### High level
+
+**TODO** combine this with the below section
+
+The Converter does two transformations.
+This first is convert the token based on a price oracle. (For example,
+if we are sent 1000 JUNO, we convert to 1200 OSMO based on some oracle, for example
+TWAP feed updated daily).
+
+The second step is to apply a discount, which captures both the volatility of the remote asset
+(discount should be greater than any expected change between oracle feed updates), as well as
+a general feeling that $1000 of remote assets shouldn't have the same power as $1000 of native
+assets. For example, the Converter is configured with a 50% discount for the JUNO provider,
+and thus those 1000 JUNO actually turn into 600 OSMO of "virtual stake".
+
+By itself, a Converter cannot impact the local staking system.  It must connect to the _Meta-Staking_ system,
+which will convert the "virtual stake" into actual stake in the dPoS system, and produce the rewards as well.
+The _Meta-Staking_ system is a permissioned contract that can access custom Cosmos SDK functionality
```

Comment:
```suggestion
The _Virtual-Staking_ system is a permissioned contract that can access custom Cosmos SDK functionality
```

Reply:
Believe there is some flexibility on the implementation here, it could be a permissioned contract or a custom SDK module.

Reply:
We need to define a clear interface for the Converter to call the VirtualStaking, to allow swapping out such VirtualStaking.

* It could be a contract doing a lot of work, then calling a few custom SDK function.
* It could be a contract as a thin wrapper to call into custom SDK functions
* It could be an entry point to a precompile (as Composable prefers "magic addresses" to CustomMsg)

 But if they all fit the same interface, we don't need to change the converter logic, or retest the ibc handling when swapping this out.

I will explain a bit more on this in the doc